netkan/netkan/spacedock_adder.py

```diff
@@ -0,0 +1,107 @@
+                to_delete = []
+                for msg in messages:
+                    if self.try_add(json.loads(msg.body)):
+                        # Successfully handled -> OK to delete
+                        to_delete.append(self.deletion_msg(msg))
+                self.queue.delete_messages(Entries=to_delete)
+                # Clean up GitPython's lingering file handles between batches
+                self.netkan_repo.close()
+
+    def try_add(self, info):
+        netkan = self.make_netkan(info)
+
+        # Create .netkan file or quit if already there
+        netkan_path = Path(self.netkan_repo.working_dir,
+                           'NetKAN', f'{netkan.identifier}.netkan')
+        if netkan_path.exists():
+            # Already exists, we are done
+            return True
+        # Otherwise create
+        netkan_path.write_text(json.dumps(netkan))
+
+        # Create branch
+        branch_name = f'add-{netkan.identifier}'
+        self.netkan_repo.remotes.origin.fetch(branch_name)
+        if branch_name not in self.netkan_repo.heads:
+            self.netkan_repo.create_head(
+                branch_name,
+                getattr(
+                    self.netkan_repo.remotes.origin.refs,
+                    branch_name
+                )
+            )
+        # Checkout branch
+        self.netkan_repo.heads[branch_name].checkout()
+
+        # Add netkan to branch
+        self.netkan_repo.index.add([netkan_path.as_posix()])
+
+        # Commit
+        self.netkan_repo.index.commit(
+            f'Add {info.name} from {info.site_name}'
+            + f'\n\nThis is an automated commit on behalf of {info.username}',
+            author=git.Actor(info.username, info.email)
+        )
+
+        # Push branch
+        self.netkan_repo.remotes.origin.push('{mod}:{mod}'.format(mod=branch_name))
+
+        # Create pull request
+        self.github_pr.create_pull_request(
+            title=f'Add {info.name} from {info.site_name}',
+            branch=branch_name,
+            body=self.get_pr_body(info)
+        )
+        return True
+
+    def make_netkan(self, info):
+        return {
+            'spec_version': 'v1.4',
+            'identifier':   re.sub(r'\W+', '', info.name),
+            '$kref':        f'#/ckan/spacedock/{info.id}',
+            'license':      info.license.replace(' ', '-'),
+            'x_via':        f'Automated {info.site_name} CKAN submission'
+        }
+
+    def get_pr_body(self, info):
+        return f"""\
+This pull request was automatically generated by {info.site_name} on behalf of {info.username}, to add [{info.name}]({info.mod_url}) to CKAN.
+Please direct questions about this pull request to [{info.username}]({info.user_url}).
+Mod details:
+    Name = {info.name}
+    Author = {info.username}
+    Abstract = {info.short_description}
+    License = {info.license}
+    Homepage = {info.external_link}
+    Description =
+{info.description}"""
```

Comment:
This might be a little neater as brackets (short of using a template of some description):

```python
    def get_pr_body(self, info):
        return (
            f'This pull request was automatically generated by {info.site_name}'
            f' on behalf of {info.username}, to add [{info.name}]({info.mod_url})'
            f' to CKAN.\n'
            f'Please direct questions about this pull request to '
            f'[{info.username}]({info.user_url}).\n'
            f'Mod details:\n'
            f'    Name = {info.name}\n'
            f'    Author = {info.username}\n'
            f'    Abstract = {info.short_description}\n'
            f'    License = {info.license}\n'
            f'    Homepage = {info.external_link}\n'
            f'    Description = {info.description}'
        )
```

Reply:
Switched to `string.Template`.

Reply:
Neat, we might want to put some thought around the templates at some point, but this is a good start.